### DISCUSSION TITLE
Date-time-stamp in backup file name

### DISCUSSION THREAD

**Discussion by 4O4A** [magicbug/Cloudlog#2052] [Ideas / Feature Requests]
Hi guys,

Is it possible to include a date/timestamp in a backup file name?
I'm asking for a friend who overwrite good backup with wrong one.

Regards,
Dragan, 4O4A

**Comment by phl0** [comment]
Of course thats possible. I wrote a PoC here: https://github.com/magicbug/Cloudlog/commit/b6f35cfd2a904c549f9d07128d7061cac4d1d48b.
The thing is that scripts that automatically download the backups will fail because they would have to grep the filename from the output after the backup has been generated. Also the backup directory will slowly start to fill because each backup gets its own file whereas in the current code the older backup is simply overwritten by the current one. So server admins will have to clean up here from time to time.
I would like to hear opinions from @AndreasK79 and @magicbug before making this a PR.

**Comment by AndreasK79** [comment]
I think the cost of space is worth it against overwriting a good backup. You could always make a script that prunes older backups if space is a problem.

**Reply by phl0** [reply]
> I think the cost of space is worth it against overwriting a good backup. You could always make a script that prunes older backups if space is a problem.

Yeah. My only concern is that cron job which regularly downloads the backup will have to be modified so that it gets the recent filename before downloading. On the pro side: It is a little more safe because one cannot simply download a backup of other users.

**Comment by AndreasK79** [comment]
@4O4A In the latest dev, the backup adds date and time in the filename. It can also be set up to run via cron with and API key.

**Comment by AndreasK79** [comment]
Closing as I have integrated this into dev.
